### PR TITLE
Convert the cloudinary package to TypeScript

### PR DESCRIPTION
**Pull request by timleslie**
Converts the remaining JavaScript code in the `cloudinary` package to TypeScript. I've confirmed that the tests still pass 👍

### CODE REVIEW

File: packages-next/cloudinary/src/Implementation.ts [hunk 1]
```diff
@@ -1,10 +1,63 @@
+import fs from 'fs';
 import cuid from 'cuid';
-import { PrismaFieldAdapter } from '@keystone-next/adapter-prisma-legacy';
+import { PrismaFieldAdapter, PrismaListAdapter } from '@keystone-next/adapter-prisma-legacy';
 import { Implementation } from '@keystone-next/fields';
+import { FieldConfigArgs, FieldExtraArgs } from '@keystone-next/fields/src/Implementation';
```
Comment (mitchellhamilton) [inline packages-next/cloudinary/src/Implementation.ts:5]: https://github.com/keystonejs/keystone/pull/5415/files#r611261036

**Review by emmatown — CHANGES_REQUESTED** [review]

File: packages-next/cloudinary/src/Implementation.ts [hunk 2]
```diff
@@ -1,10 +1,62 @@
...
+  id: string;
+  filename: string;
+  originalFilename: string;
+  mimetype: any;
+  encoding: any;
+  _meta: cloudinary.UploadApiResponse;
+};
+
+export type CloudinaryImageFormat = {
+  prettyName?: string;
```
Comment (mitchellhamilton) [inline packages-next/cloudinary/src/Implementation.ts:20]: All of these need `| null`

File: packages-next/cloudinary/src/Implementation.ts [hunk 3]
```diff
@@ -93,31 +145,40 @@ class CloudinaryImage extends Implementation {
...
 
-  async resolveInput({ resolvedData, existingItem }) {
-    const previousData = existingItem && existingItem[this.path];
+  async resolveInput({
+    resolvedData,
+    existingItem,
+  }: {
+    resolvedData: Record<
+      P,
+      { createReadStream: () => fs.ReadStream; filename: string; mimetype: any; encoding: any }
```
Comment (mitchellhamilton) [inline packages-next/cloudinary/src/Implementation.ts:177]: This type should be `FileUpload` from `'graphql-upload'`

Comment (timleslie) [reply packages-next/cloudinary/src/Implementation.ts:177]: Ah, yes, thank you!

**Review by emmatown — APPROVED** [review]

**Pull request merged**